pkg/devicerepository/store/remote/schema.go

```diff
@@ -0,0 +1,436 @@
+	Rx2Frequency              float64             `yaml:"rx2Frequency"`
+	FactoryPresetFrequencies  []float64           `yaml:"factoryPresetFrequencies"`
+	MaxEIRP                   float32             `yaml:"maxEIRP"`
+	MaxDutyCycle              float64             `yaml:"maxDutyCycle"`
+	Supports32BitFCnt         bool                `yaml:"supports32bitFCnt"`
+}
+
+var (
+	errUnknownBand = errors.DefineNotFound("unknown_band", "unknown band `{band_id}`")
+	errNoProfile   = errors.DefineNotFound("no_profile", "device does not support region `{region}`")
+
+	errUnknownMACVersion                = errors.DefineNotFound("unknown_mac_version", "unknown LoRaWAN version `{mac_version}`")
+	errUnknownRegionalParametersVersion = errors.DefineNotFound("unknown_regional_parameters_version", "unknown Regional Parameters version `{phyVersion}`")
+)
+
+// ToTemplatePB returns a ttnpb.EndDeviceTemplate from an end device profile.
+func (p EndDeviceProfile) ToTemplatePB(ids *ttnpb.EndDeviceVersionIdentifiers, info *ttnpb.EndDeviceModel_FirmwareVersion_Profile) (*ttnpb.EndDeviceTemplate, error) {
+	phyVersion, ok := regionalParametersToPB[p.RegionalParametersVersion]
+	if !ok {
+		return nil, errUnknownRegionalParametersVersion.WithAttributes("phyVersion", p.RegionalParametersVersion)
+	}
+
+	paths := []string{
+		"version_ids",
+		"supports_join",
+		"supports_class_b",
+		"supports_class_c",
+		"lorawan_version",
+		"lorawan_phy_version",
+	}
+	dev := ttnpb.EndDevice{
+		VersionIDs:        ids,
+		SupportsJoin:      p.SupportsJoin,
+		SupportsClassB:    p.SupportsClassB,
+		SupportsClassC:    p.SupportsClassC,
+		LoRaWANVersion:    p.MACVersion,
+		LoRaWANPHYVersion: phyVersion,
+	}
+
+	if info.CodecID != "" {
+		dev.Formatters = &ttnpb.MessagePayloadFormatters{
+			DownFormatter: ttnpb.PayloadFormatter_FORMATTER_REPOSITORY,
+			UpFormatter:   ttnpb.PayloadFormatter_FORMATTER_REPOSITORY,
+		}
+		paths = append(paths, "formatters")
+	}
+
+	dev.MACSettings = &ttnpb.MACSettings{}
+	if p.ClassBTimeout > 0 {
+		t := time.Duration(p.ClassBTimeout) * time.Second
+		dev.MACSettings.ClassBTimeout = &t
+		paths = append(paths, "mac_settings.class_b_timeout")
+	}
+	if p.ClassCTimeout > 0 {
+		t := time.Duration(p.ClassCTimeout) * time.Second
+		dev.MACSettings.ClassCTimeout = &t
+		paths = append(paths, "mac_settings.class_c_timeout")
+	}
+	if p.PingSlotDataRateIndex > 0 {
+		dev.MACSettings.PingSlotDataRateIndex = &ttnpb.DataRateIndexValue{
+			Value: p.PingSlotDataRateIndex,
+		}
+		paths = append(paths, "mac_settings.ping_slot_data_rate_index")
+	}
+	if p.PingSlotFrequency > 0 {
+		dev.MACSettings.PingSlotFrequency = &pbtypes.UInt64Value{
+			Value: uint64(p.PingSlotFrequency * 100000),
+		}
+		paths = append(paths, "mac_settings.ping_slot_frequency")
+	}
+	if p.PingSlotPeriod > 0 {
+		dev.MACSettings.PingSlotPeriodicity = &ttnpb.PingSlotPeriodValue{
+			Value: pingSlotPeriodToPB[p.PingSlotPeriod],
```

Comment:
Where does this value come from? Using seconds here is not the standard "LoRaWAN way"

Reply:
For context, The `PingSlotPeriod` value used here is coming from the Device Repository where it is validated against [the following schema](https://lorawan-schema.org/draft/devices/1):

```
    "pingSlotPeriod": {
      "enum": [32, 64, 128, 256, 512, 1024, 2048, 4096],
      "description": "Ping slot period (seconds)"
    },
```

In #3433, we decided that values read by the device repository can be assumed to comply with the schema. Is there any better to go about with this?

Reply:
For more context, this comes from the LoRaWAN Backend Interfaces' Device Profile

Reply:
The spec just specifies the value type as "number", are you sure the value is meant to be in seconds?